src/aiida_quantumespresso/utils/hubbard.py

```diff
@@ -313,6 +312,243 @@ def get_hubbard_for_supercell(self, supercell: StructureData, thr: float = 1e-3)
 
         return HubbardStructureData.from_structure(structure=supercell, hubbard=new_hubbard)
 
+    def get_interacting_pairs(self) -> Dict[str, List[str]]:
+        """Return tuple of kind name interaction pairs.
+
+        :returns: dictionary of onsite kinds with a list of V kinds
+        """
+        pairs_dict = dict()
+        sites = self.hubbard_structure.sites
+        parameters = self.hubbard_structure.hubbard.parameters
+
+        onsite_indices = [p.atom_index for p in parameters if p.atom_index == p.neighbour_index]
+
+        for index in onsite_indices:
+            name = sites[index].kind_name
+            if name not in pairs_dict:
+                pairs_dict[name] = []
+
+        for parameter in parameters:
+            onsite_name = sites[parameter.atom_index].kind_name
+            neigh_name = sites[parameter.neighbour_index].kind_name
+
+            if onsite_name in pairs_dict and onsite_name != neigh_name:
+                if not neigh_name in pairs_dict[onsite_name]:
+                    pairs_dict[onsite_name].append(neigh_name)
+
+        return pairs_dict
+
+    def get_pairs_radius(
+        self,
+        onsite_index: int,
+        neighbours_names: List[str],
+        number_of_neighbours: int,
+        radius_max: float = 7.0,
+        thr: float = 1.0e-2,
+    ) -> Tuple[float, float]:
+        """Return the minimum and maximum radius of the first neighbours of the onsite site.
+
+        :param onsite_index: index in the structure of the onsite Hubbard atom
+        :param neighbours_names: kind names of the neighbours
+        :param number_of_neighbours: number of neighbours coming to select
+        :param radius_max: maximum radius (in Angstrom) to use for looking for neighbours
+        :param thr: threshold (in Angstrom) for defining the shells
+        :return: (radius min +thr, radius max -thr) defining the shells containing only the first neighbours
+        """
+        rmin = 0
+        pymat = self.hubbard_structure.get_pymatgen_structure()
+        _, neigh_indices, _, distances = pymat.get_neighbor_list(sites=[pymat[onsite_index]], r=radius_max)
```

Comment:
'get_pymatgen_structure' , could be problematic if the HubbardStructureData is 1D, or 2D

Reply:
I am start wondering whether we should change the aiida-core implementation of `get_pymatgen_structure` instead. @sphuber, @mbercx opinions?

Reply:
Ok, having a look into the `get_pymatgen_structure` there is a strict [check](https://github.com/aiidateam/aiida-core/blob/8a286f26e8d303c498ac2eabd49be5f1f4ced9ef/src/aiida/orm/nodes/data/structure.py#L1774). This means that any pbc different from 3D would raise error. But as far as I saw in pymatgen core modules, they actually allow for different pbc (e.g. the [lattice](https://github.com/materialsproject/pymatgen/blob/0a1035bf9bde5ceb9e55a8a8086f5eef5656953a/pymatgen/core/lattice.py#L34) class allows any pbc).

So one should define a `Lattice` instance instead of passing directly the `self.cell`, which is simply array like. The `Lattice` class also provides `pbc` in input, so it should be straightforward.

I don't know about older version of pymatgen, maybe they didn't have this class before? Or maybe it wasn't realised at the time?

I would simply vote to do this fix in aiida-core, since this method is used also in other codes (e.g. aiida-vibroscopy, and who know what else), and it wouldn't make sense to patch this all over the places, since then also who knows what happens if settings pbc=3D in pymatgen for a 2D system.

@AndresOrtegaGuerrero we can simply open a simple PR on aiida-core and have the fix there, and you can test it. If you understood, this should be a very easy fix, so you can try it yourself as well.

Reply:
Alright, just opened the PR on aiida-core. @AndresOrtegaGuerrero it would be good if you tried to pull the new changes of aiida-core from [this PR](https://github.com/aiidateam/aiida-core/pull/6281), and see if it solves also all the other issues raised in the other PR #998 of aiida-quantumespresso you opened.

Reply:
I am trying to switch to the latest aiida-core, but i keep getting a problem in the migration 

Traceback (most recent call last):
  File "/Users/ogaa/anaconda3/envs/aiida2_3/bin/verdi", line 8, in <module>
    sys.exit(verdi())
  File "/Users/ogaa/anaconda3/envs/aiida2_3/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/ogaa/anaconda3/envs/aiida2_3/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/ogaa/anaconda3/envs/aiida2_3/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/ogaa/anaconda3/envs/aiida2_3/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/ogaa/anaconda3/envs/aiida2_3/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/ogaa/anaconda3/envs/aiida2_3/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/ogaa/Documents/aiida-core/aiida/cmdline/commands/cmd_storage.py", line 78, in storage_migrate
    storage_cls.migrate(profile)
  File "/Users/ogaa/Documents/aiida-core/aiida/storage/psql_dos/backend.py", line 92, in migrate
    migrator.migrate()
  File "/Users/ogaa/Documents/aiida-core/aiida/storage/psql_dos/migrator.py", line 368, in migrate
    branches = self._alembic_script().revision_map.get_revision(version).branch_labels
  File "/Users/ogaa/anaconda3/envs/aiida2_3/lib/python3.9/site-packages/alembic/script/revision.py", line 574, in get_revision
    return self._revision_for_ident(resolved, branch_label)
  File "/Users/ogaa/anaconda3/envs/aiida2_3/lib/python3.9/site-packages/alembic/script/revision.py", line 622, in _revision_for_ident
    raise ResolutionError(
alembic.script.revision.ResolutionError: No such revision or branch 'main_0002'